src/main.rs

```diff
@@ -1,3 +1,157 @@
+struct Mips {
+    regs: [u32; 32],
+    floats: [f32; 32],
+    mult_hi: u32,
+    mult_lo: u32,
+    pc: usize,
+
+    // Enough for six MIPS instructions.
+    memory: [u32; 13]
+}
+
+struct Rtype {
+    rs: usize,
+    rt: usize,
+    rd: usize,
+    shamt: u8,
+    funct: u8
+}
```

Comment:
Note for later: we should just merge our impls.
Also nitpick but we shouldn't use a `usize` for something that isn't denoting a size

Reply:
I'm still unconvinced that writing ``u5`` and ``u6`` types was a good idea. I'd prefer not to implement them into my codebase. Also, keeping the assembler and disassembler decoupled feels worth having a more in-depth discussion over.

re: usize: I can change that.